powerstore/resource_snapshotrule.go

```diff
@@ -0,0 +1,352 @@
+			},
+			"managed_by_id": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The unique id of the managing entity.",
+				MarkdownDescription: "The unique id of the managing entity.",
+			},
+		},
+	}, nil
+}
+
+// NewResource is a wrapper around provider
+func (r resourceSnapshotRuleType) NewResource(_ context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return resourceSnapshotRule{
+		p: *(p.(*Pstoreprovider)),
+	}, nil
+}
+
+type resourceSnapshotRule struct {
+	p Pstoreprovider
+}
+
+func (r resourceSnapshotRule) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider hasn't been configured before apply, likely because it depends on an unknown value from another resource. This leads to weird stuff happening, so we'd prefer if you didn't do that. Thanks!",
+		)
+		return
+	}
+
+	log.Printf("Started Creating Snapshot Rule")
+	var plan models.SnapshotRule
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	snapshotRuleCreate := planToSnapshotRuleParam(plan)
+
+	log.Printf("Calling api to create snapshotrule")
+
+	// Create New SnapshotRule
+	// The function returns only ID of the newly created snapshot rule
+	createRes, err := r.p.client.PStoreClient.CreateSnapshotRule(context.Background(), snapshotRuleCreate)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating snapshot rule",
+			"Could not create snapshot rule, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	log.Printf("Calling api to get snapshotrule created info")
+
+	// Get SnapshotRule Details using ID retrieved above
+	getRes, err := r.p.client.PStoreClient.GetSnapshotRule(context.Background(), createRes.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting snapshot rule after creation",
+			"Could not get snapshot rule, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	state := models.SnapshotRule{}
+	updateSnapshotRuleState(&state, getRes)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	log.Printf("Successfully done with Create")
+}
+
+// Read fetch info about asked snapshot rule
+func (r resourceSnapshotRule) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+
+	var state models.SnapshotRule
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get snapshot details from API and then update what is in state from what the API returns
+	id := state.ID.Value
+	response, err := r.p.client.PStoreClient.GetSnapshotRule(context.Background(), id)
+
+	// todo distnguish whether error is for resource presence, in case resource is not present
```

Comment:
comment required?

Reply:
yes, this is to let us know that this is to be done